.github/workflows/test-and-publish.yaml

```diff
@@ -128,8 +128,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        # The lowest supported version is Ubuntu 20.04 + Python 3.8 or macOS 12 + Python 3.9
-        os: [ 'ubuntu-20.04', 'macos-12', 'macos-14', 'windows-2022' ]
+        os: [ 'ubuntu-20.04', 'ubuntu-24.04', 'macos-12', 'macos-14', 'windows-2022' ]
```

Comment:
The `build-and-test` doesn't run `setup.sh`.

Reply:
weird you just wrote that it does here in slack:


Tom Tang  3:30 PM
I meant your added CI tests don't test your changes, which should be done in the same PR. (edited) 

Philippe Laporte  3:31 PM
are you saying CI does not run setup.sh?
that would be quite bad

Tom Tang  3:32 PM
yes during the build-and-test steps (edited) 

Philippe Laporte  3:33 PM
ok so we're good

Reply:
got it, thanks